### PR TITLE
Merge branch Yzm1234 to master branch

### PR DESCRIPTION
**Pull request by Yzm1234**

### CODE REVIEW

File: test/data/a.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+xyz1
```
Comment (Yzm1234) [inline test/data/a.txt:1]: Add general strings to make a.txt file pass uniitests( previously it was empty so lead to failure.)

File: test/data/b.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+xyz2
```
Comment (Yzm1234) [inline test/data/b.txt:1]: Add general strings to make b.txt file pass uniitests( previously it was empty so lead to failure.)

File: .github/workflows/action.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    - name: Set up test environment
+      if: "!contains(github.event.head_commit.message, 'skip ci')"
+      shell: bash
+      env:
+        KBASE_TEST_TOKEN: ${{ secrets.KBASE_TEST_TOKEN}}
+      run: |
+        # Verify kb_sdk_actions clone worked
+        test -f "$HOME/kb_sdk_actions/bin/kb-sdk" && echo "CI files cloned"
+        # Pull kb-sdk & create startup script
+        docker pull kbase/kb-sdk
```
Comment (Yzm1234) [inline .github/workflows/action.yml]: changing action to kb_adk_actions

**Review by Yzm1234 — COMMENTED** [review]
adding comments to changes for PR

File: .github/workflows/action.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+name: KBase SDK Tests
+
+on:
```
Comment (ialarmedalien) [inline .github/workflows/action.yml]: This can be changed to
```
on: [push, pull_request]
```

i.e. remove
```
  push:
  pull_request:
```

Comment (Yzm1234) [reply .github/workflows/action.yml]: ok, will do it!

File: test/KBaseReport_server_test.py [hunk 5]
```diff
@@ -13,6 +13,8 @@
 from installed_clients.DataFileUtilClient import DataFileUtil
 from installed_clients.WorkspaceClient import Workspace
 from TemplateUtil_test import get_test_data
+from installed_clients.baseclient import ServerError
+from installed_clients.baseclient import ServerError
```
Comment (ialarmedalien) [inline test/KBaseReport_server_test.py]: Duplicated line

File: test/KBaseReport_server_test.py [hunk 6]
```diff
@@ -58,12 +60,18 @@ def setUpClass(cls):
...
+                'file_path': cls.a_file_path, 'make_handle': 0
+            })
+        except Exception as e:
+            print('Unable to store ' + cls.a_file_path + str(e))
+        try:
+            cls.b_file_shock = cls.dfu.file_to_shock({
+                'file_path': cls.b_file_path, 'make_handle': 0
+            })
+        except Exception as e:
+            print('Unable to store ' + cls.b_file_path + str(e))
```
Comment (ialarmedalien) [inline test/KBaseReport_server_test.py]: There are a load of duplicated commands that could be replaced with a `for` loop that uses `a` and `b`. Could you do a bit of refactoring here to get rid of the duplication?

**Review by ialarmedalien — APPROVED** [review]
Looks great! Thanks.

**Pull request merged**